src/main/java/com/project/stockexchangeappbackend/service/StockServiceImpl.java

```diff
@@ -16,28 +17,49 @@
 @RequiredArgsConstructor
 public class StockServiceImpl implements StockService {
 
-    private final StockRepository repository;
+    private final StockRepository stockRepository;
 
     @Override
     @LogicBusinessMeasureTime
     @Transactional(readOnly = true)
     public Stock getStockById(Long id) {
-        return repository.findById(id).orElseThrow(() -> new EntityNotFoundException("Stock Not Found"));
+        return stockRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("Stock Not Found"));
     }
 
     @Override
     @LogicBusinessMeasureTime
     @Transactional(readOnly = true)
     public Page<Stock> getStocks(Pageable pageable, Specification<Stock> specification) {
-        return repository.findAll(specification, pageable);
+        return stockRepository.findAll(specification, pageable);
     }
 
     @Override
     @LogicBusinessMeasureTime
+    @Transactional
     public Stock getStockByAbbreviation(String abbreviation) {
-        return repository.findByAbbreviationIgnoreCase(abbreviation).orElseThrow(() ->
+        return stockRepository.findByAbbreviationIgnoreCase(abbreviation).orElseThrow(() ->
                 new EntityNotFoundException("Stock Not Found"));
     }
 
+    @Override
+    @LogicBusinessMeasureTime
+    public void updateStock(StockDTO stockDTO, String id) {
+        Stock stock = getStockByIdOrAbbreviation(id);
+        stock.setAmount(stockDTO.getAmount());
```

Comment:
What if it will be set to negative number? Or when sum of possessed by users amount of stock is greater or less than this value?